### PR TITLE
Fix xz guides not showing when translating with Shift

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by huffman** [comment]
# Test Plan

Note: For the following tests, please also test that the x/z snapping is enabled and the lines show up when you press the shift key WITHOUT moving the mouse.
  1. https://highfidelity.testrail.net/index.php?/tests/view/70779
  1. https://highfidelity.testrail.net/index.php?/tests/view/70810

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11423/HighFidelity-Beta-PR13648-25b558e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11423/HighFidelity-Beta-PR13648-25b558e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11423/HighFidelity-Beta-Interface-PR13648-25b558e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11423/HighFidelity-Beta-PR13648-25b558e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11423/HighFidelity-Beta-Interface-PR13648-25b558e.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
@huffman I am unable to find you on slack....

I want to clarify that when I select an entity with create, holding shift will show the lines even without moving the object or the mouse? I am currently not seeing that. I have to move the cube or entity to see the lines. Wanted to clarify this is indeed a fail.

**Comment by huffman** [comment]
Hi qlimnavarro,

Sorry, my description was unclear. The steps would be something like:

Select entity
Click on entity to begin dragging
While mouse is stationary, press shift. You should see it switch to the xz
snapping mode with the lines.

Please let me know if that doesn't make sense. Previously you would need to
move the mouse again to see it switch to the snapping mode.

**Comment by qlimnavarro** [comment]
Pass @qliacoombs @Qliemillar 
Lines appear and snap well.

**Comment by huffman** [comment]
Thank you @qlimnavarro

File: scripts/system/libraries/entitySelectionTool.js [hunk 1]
```diff
@@ -1768,10 +1833,46 @@ SelectionDisplay = (function() {
                     vector.x = 0;
                 }
                 if (!isConstrained) {
+                    var xStart = Vec3.sum(startPosition, {
+                        x: -10000,
```
Comment (thoys) [inline scripts/system/libraries/entitySelectionTool.js]: What does `-10000` mean? `-RAIL_AXIS_LENGTH`? Define constant please

File: scripts/system/libraries/entitySelectionTool.js [hunk 2]
```diff
@@ -1768,10 +1833,46 @@ SelectionDisplay = (function() {
...
+                        y: 0,
+                        z: 0
+                    });
+                    var xEnd = Vec3.sum(startPosition, {
+                        x: 10000,
```
Comment (thoys) [inline scripts/system/libraries/entitySelectionTool.js]: What does `10000` mean? `RAIL_AXIS_LENGTH`? Define constant please

File: scripts/system/libraries/entitySelectionTool.js [hunk 3]
```diff
@@ -1768,10 +1833,46 @@ SelectionDisplay = (function() {
...
+                        y: 0,
+                        z: 0
+                    });
+                    var zStart = Vec3.sum(startPosition, {
+                        x: 0,
+                        y: 0,
+                        z: -10000
```
Comment (thoys) [inline scripts/system/libraries/entitySelectionTool.js]: What does `-10000` mean? `-RAIL_AXIS_LENGTH`? Define constant please

File: scripts/system/libraries/entitySelectionTool.js [hunk 4]
```diff
@@ -1768,10 +1833,46 @@ SelectionDisplay = (function() {
...
+                    });
+                    var zEnd = Vec3.sum(startPosition, {
+                        x: 0,
+                        y: 0,
+                        z: 10000
```
Comment (thoys) [inline scripts/system/libraries/entitySelectionTool.js]: What does `10000` mean? `RAIL_AXIS_LENGTH`? Define constant please

File: scripts/system/libraries/entitySelectionTool.js [hunk 5]
```diff
@@ -873,11 +911,14 @@ SelectionDisplay = (function() {
     };
 
     // FUNCTION: MOUSE MOVE EVENT
+    var lastMouseEvent = null;
     that.mouseMoveEvent = function(event) {
+		print("Got mouse move event", JSON.stringify(event));
```
[the remark this answers is not in the document]
Comment (huffman) [reply scripts/system/libraries/entitySelectionTool.js]: Oops, I did not mean to leave that. Thanks

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11462/HighFidelity-Beta-PR13648-ae26c5f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11462/HighFidelity-Beta-PR13648-ae26c5f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11462/HighFidelity-Beta-Interface-PR13648-ae26c5f.dmg

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @qliacoombs @conklin94122 
Test cases pass and lines appear without moving mouse as expected.

**Pull request merged**